### PR TITLE
Fix zipfile target.

### PR DESCRIPTION
**Pull request by pillarsdotnet**
See https://github.com/jordansissel/fpm/issues/1313

### CODE REVIEW

**Comment by pillarsdotnet** [comment]
Dunno why travis-ci is failing, but a trivial change (adding a line to README.rst) also fails.

File: lib/fpm/package/zip.rb [hunk 1]
```diff
@@ -36,28 +36,12 @@ def input(input_path)
...
-        return "-j"
-      when /\.tar\.gz$|\.tgz$/
-        return "-z"
-      when /\.tar\.xz$/
-        return "-J"
-      else
-        return nil
+    realpath = Pathname.new(output_path).realdirpath.to_s
+    ::Dir.chdir(staging_path) do
+      safesystem("zip", "-9ry", realpath, ".")
```
Comment (jordansissel) [inline lib/fpm/package/zip.rb]: This is a behavioral change (adding -r and -y flags). What is the expected impact on users? I imagine the `-y` flag to build the zip will not work when extracting the resulting zip on Windows?

**Comment by jordansissel** [comment]
Travis is failing due to an integration problem between osx and fpm's test suite. I ignore this particular error.

File: lib/fpm/package/zip.rb (hunk 1 above)
Comment (jordansissel) [inline lib/fpm/package/zip.rb]: The reason I mention this is that the PR and linked issue focus on fixing the directory path problem, and this is fixed by other parts of the patch (the two previous lines (`realpath` and `Dir.chdir`), and adding `-9ry` seems unrelated to the bug fix.

Comment (pillarsdotnet) [inline lib/fpm/package/zip.rb]: the "-r" is necessary; the "-y" is helpful on unix but (...testing...) sadly unsupported on Windows.  I'll re-roll the patch with out the "-y" flag.

**Comment by bazimov** [comment]
Anything blocking this MR, we are seeing same issue on zip functionality. This MR fixes that issue.

**Comment by jordansissel** [comment]
I'd like to see more tests (who wouldn't! hehe) for the zip output, but I won't block this PR on that.

Thank you for helping make fpm more awesome :)
